src/shared/config.ts

```diff
@@ -24,60 +24,145 @@
+}
 
-  PARTICIPANT_ID: string;
+interface DbPool {
+  min: number;
+  max: number;
+  acquireTimeoutMillis: number;
+  createTimeoutMillis: number;
+  destroyTimeoutMillis: number;
+  idleTimeoutMillis: number;
+  reapIntervalMillis: number;
+  createRetryIntervalMillis: number;
+}
+
+interface DatabaseConfig {
+  client: string;
+  version?: string;
+  useNullAsDefault?: boolean;
+  connection: DbConnection | string;
+  pool?: DbPool;
 
-  // ../lib/config.
-  DB_ENVIRONMENT: string;
-  DATABASE?: {
-    ACQUIRE_TIMEOUT_MILLIS: number;
-    CREATE_RETRY_INTERVAL_MILLIS: number;
-    CREATE_TIMEOUT_MILLIS: number;
-    DEBUG: boolean;
-    DESTROY_TIMEOUT_MILLIS: number;
-    DIALECT: string;
-    HOST: string;
-    IDLE_TIMEOUT_MILLIS: number;
-    PASSWORD: string;
-    POOL_MAX_SIZE: number;
-    POOL_MIN_SIZE: number;
-    PORT: number;
-    REAP_INTERVAL_MILLIS: number;
-    SCHEMA: string;
-    USER: string;
+  migrations: {
+    directory: string;
+    tableName: string;
+    stub?: string;
+    loadExtensions: string[];
   };
 
-  MIGRATIONS?: {
-    DISABLED: boolean;
-    RUN_DATA_MIGRATIONS: boolean;
+  seeds: {
+    directory: string;
+    loadExtensions: string[];
   };
+}
 
-  // inspect.ts
-  INSPECT?: {
-    DEPTH?: number;
-    SHOW_HIDDEN?: boolean;
-    COLOR?: boolean;
+interface ServiceConfig {
+  PORT: number;
+  HOST: string;
+  PARTICIPANT_ID: string;
+  DATABASE?: DatabaseConfig;
+  SHOULD_USE_IN_MEMORY_DB: string;
+  INSPECT: {
+    DEPTH: number;
+    SHOW_HIDDEN: boolean;
+    COLOR: boolean;
   };
 }
 
-const RC = parse(rc('AS', Config)) as ServiceConfig
+const ConvictConfig = Convict<ServiceConfig>({
+  SHOULD_USE_IN_MEMORY_DB: {
+    doc: 'Should we use the in memory database',
+    format: ['true', 'false'],
+    default: 'false',
+    env: 'SHOULD_USE_IN_MEMORY_DB'
+  },
+  HOST: {
+    doc: 'The Hostname/IP address to bind.',
+    format: '*',
+    default: '0.0.0.0',
+    env: 'HOST',
+    arg: 'host'
+  },
+  PORT: {
+    doc: 'The port to bind.',
+    format: 'port',
+    default: 4004,
+    env: 'PORT',
+    arg: 'port'
+  },
+  PARTICIPANT_ID: {
+    doc: 'Service ID for the Mojaloop network.',
+    format: String,
+    default: 'auth-service',
+    env: 'PARTICIPANT_ID',
+    arg: 'participantId'
+  },
+  INSPECT: {
+    DEPTH: {
+      doc: 'Inspection depth',
+      format: 'nat',
+      env: 'INSPECT_DEPTH',
+      default: 4
+    },
+    SHOW_HIDDEN: {
+      doc: 'Show hidden properties',
+      format: 'Boolean',
+      default: false
+    },
+    COLOR: {
+      doc: 'Show colors in output',
+      format: 'Boolean',
+      default: true
+    }
+  }
+})
+
+// Load and validate database config
+const shouldUseInMemoryDB = ConvictConfig.get('SHOULD_USE_IN_MEMORY_DB') === 'true'
+let DBConfig: DatabaseConfig = DbConfig.development
+
+if (shouldUseInMemoryDB) {
+  DBConfig = DbConfig.test
+}
+
+// Load and validate general config
+ConvictConfig.loadFile(`${__dirname}/../../config/config.json`)
```

Comment:
there is a need for separate config files for every environment as it is done here: https://github.com/mojaloop/thirdparty-scheme-adapter/blob/master/src/shared/config.ts#L284-L286